cirq/experiments/t2_decay_experiment.py

```diff
@@ -0,0 +1,336 @@
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class ExperimentType(enum.Enum):
+    RAMSEY = 1  # Often denoted as t2*
+    HAHN_ECHO = 2  # Spin echo or t2
+    CPMG = 3  # Carr-Purcell-Meiboom-Gill sequence
+
+
+_T2_COLUMNS = ['delay_ns', 0, 1]
+
+
+def t2_decay(sampler: work.Sampler,
+             *,
+             qubit: devices.GridQubit,
+             experiment_type: 'ExperimentType' = ExperimentType.RAMSEY,
+             num_points: int,
+             max_delay: 'cirq.DURATION_LIKE',
+             min_delay: 'cirq.DURATION_LIKE' = None,
+             repetitions: int = 1000) -> 'cirq.experiments.T2DecayResult':
+    """Runs a t2 transverse relaxation experiment.
+
+    Initializes a qubit into a superposition state, evolves the system using
+    rules determined by the experiment type and by the delay parameters,
+    then rotates back for measurement.  This will measure the phase decoherence
+    decay.  This experiment has three types of T2 metrics, each which measure
+    a different slice of the noise spectrum.
+
+    For the Ramsey experiment type (often denoted T2*), the state will be
+    prepared with a square root Y gate (`cirq.Y ** 0.5`) and then waits for
+    a variable amount of time.  After this time, it will do basic state
+    tomography to measure the expectation of the Pauli-X and Pauli-Y operators
+    by performing either a `cirq.Y ** -0.5` or `cirq.X ** -0.5`.  The square of
+    these two measurements is summed to determine the length of the Bloch
+    vector. This experiment measures the phase decoherence of the system under
+    free evolution.
+
+    For the Hahn echo experiment (often denoted T2 or spin echo), the state
+    will also be prepared with a square root Y gate (`cirq.Y ** 0.5`).
+    However, during the mid-point of the delay time being measured, a pi-pulse
+    (`cirq.X`) gate will be applied to cancel out inhomogeneous dephasing.
+    The same method of measuring the final state as Ramsey experiment is applied
+    after the second half of the delay period.
+
+    CPMG, or the Carr-Purcell-Meiboom-Gill sequence, is currently not
+    implemented.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        experiment_type: The type of T2 test to run.
+        num_points: The number of evenly spaced delays to test.
+        max_delay: The largest delay to test.
+        min_delay: The smallest delay to test. Defaults to no delay.
+        repetitions: The number of repetitions of the circuit
+             for each delay and for each tomography result.
+
+    Returns:
+        A T2DecayResult object that stores and can plot the data.
+    """
+    min_delay_dur = value.Duration(min_delay)
+    max_delay_dur = value.Duration(max_delay)
+
+    # Input validation
+    if repetitions <= 0:
+        raise ValueError('repetitions <= 0')
+    if repetitions % 2 != 0:
+        raise ValueError('Total repetitions must be an even number.')
```

Comment:
out of curiousity, how come?

Reply:
Whoops.  It was initially because I split the repetitions between x-basis and y-basis measurements.  It got too confusing whether things should be reps or reps/2 all over the place.

Removed this check.